src/pages/Script/ScriptsComp.tsx

```diff
@@ -122,33 +130,46 @@ export const ScriptCells = ({
 }) => {
   const history = useHistory()
   const { codeHash, hashType } = useParams<{ codeHash: string; hashType: string }>()
-
-  const cellsQuery = useQuery([`scripts_${cellType}`, codeHash, hashType, page, size], async () => {
-    const { data, meta } = await explorerService.api.requesterV2
-      .get(`scripts/${cellType}`, {
-        params: {
-          code_hash: codeHash,
-          hash_type: hashType,
-          page,
-          page_size: size,
-        },
-      })
-      .then((res: AxiosResponse) =>
-        toCamelcase<Response.Response<{ deployedCells?: CellInScript[]; referringCells?: CellInScript[] }>>(res.data),
-      )
-    const camelCellType = camelcase(cellType) as 'deployedCells' | 'referringCells'
-    if (data == null) {
-      throw new Error('Fetch Cells null')
-    }
-    const cells = data[camelCellType]!
-    if (cells == null || cells.length === 0) {
-      throw new Error('Cells empty')
-    }
-    return {
-      total: meta?.total ?? 0,
-      cells,
-    }
+  const [cellsEmpty, setCellsEmpty] = useState<Record<'deployedCells' | 'referringCells', boolean>>({
+    deployedCells: false,
+    referringCells: false,
   })
+
+  const camelCellType = camelcase(cellType) as 'deployedCells' | 'referringCells'
+
+  const cellsQuery = useQuery(
+    [`scripts_${cellType}`, codeHash, hashType, page, size],
+    async () => {
+      const { data, meta } = await explorerService.api.requesterV2
+        .get(`scripts/${cellType}`, {
+          params: {
+            code_hash: codeHash,
+            hash_type: hashType,
+            page,
+            page_size: size,
+          },
+        })
+        .then((res: AxiosResponse) =>
+          toCamelcase<Response.Response<{ deployedCells?: CellInScript[]; referringCells?: CellInScript[] }>>(res.data),
+        )
+      if (data == null) {
```

Comment:
`==` should be avoided and `===` should be used

Reply:
I think `==` can be used when checking for `null/undefined`, and `===` should be used in other cases.

Reply:
> I think `==` can be used when checking for `null/undefined`, and `===` should be used in other cases.

IMO, `==` introduces mental burden and potential misusage if this code is taken over by someone who is not familiar with this nuance.

Reply:
This should be part of the basics of JavaScript, and it is not recommended to consider this kind of scenario. Otherwise, we would also have to consider beginners who are not familiar with common content such as `const`, `destructuring syntax`, `async/await`, `Promise`, `Ternary expression`, and so on.

We should only consider limiting libraries that write tricks beyond a certain range or use unconventional methods.

Reply:
> This should be part of the basics of JavaScript, and it is not recommended to consider this kind of scenario. Otherwise, we would also have to consider beginners who are not familiar with common content such as `const`, `destructuring syntax`, `async/await`, `Promise`, `Ternary expression`, and so on.
> 
> We should only consider limiting libraries that write tricks beyond a certain range or use unconventional methods.

I prefer to eliminate bad parts of JavaScript, even though they work better in some cases, e.g., `==` covers `undefined/null`, `var` hoists variables, but they are still prone to be misused because those details are generally overwhelmed by other fashion concepts. It can be used correctly by experienced developers but may not be used well by novices.

We can define a method to match `null/undefined` with a clear function, e.g. https://github.com/lodash/lodash/blob/main/src/isNil.ts

Reply:
I agree that using `===` instead of `==` can avoid certain pitfalls and ambiguities. I also think that implicit conversions like `[]==false` are very confusing. However, `== null` is a very simple, traditional, and common case that only equals itself and `undefined`.

This is a trade-off issue. Eliminating parts of a language can reduce errors, but also takes away tools that may have legitimate uses.

We should choose suitable tools based on the actual situation and team. I think developers on our project who are unfamiliar with the traditional and basic meaning of `== null` would be an extreme minority.

So we should not increase our own development costs for this (although not by much), and using `isNil` may also introduce other readability costs.

Traditional projects and lints like Airbnb default to allowing the `== null` exception:
https://github.com/airbnb/javascript/blob/11ab37144b7f846f04f64a29b5beb6e00d74e84b/packages/eslint-config-airbnb-base/rules/best-practices.js#L54

Reply:
> I agree that using `===` instead of `==` can avoid certain pitfalls and ambiguities. I also think that implicit conversions like `[] == false` are very confusing. However, `== null` is a very simple, traditional, and common case that only equals itself and `undefined`.
> 
> This is a trade-off issue. Eliminating parts of a language can reduce errors, but also takes away tools that may have legitimate uses.
> 
> We should choose suitable tools based on the actual situation and team. I think developers on our project who are unfamiliar with the traditional and basic meaning of `== null` would be an extreme minority.
> 
> So we should not increase our own development costs for this (although not by much), and using `isNil` may also introduce other readability costs.
> 
> Traditional projects and lints like Airbnb default to allowing the `== null` exception: [airbnb/javascript@`11ab371`/packages/eslint-config-airbnb-base/rules/best-practices.js#L54](https://github.com/airbnb/javascript/blob/11ab37144b7f846f04f64a29b5beb6e00d74e84b/packages/eslint-config-airbnb-base/rules/best-practices.js#L54)

I'm convinced by the exception of `==`, let's use `==` to match `null/undefined`